mne/simulation/source.py

```diff
@@ -336,28 +340,38 @@ class SourceSimulator(object):
         The number of time samples of the simulation.
     """
 
-    def __init__(self, src, tstep=1e-3, duration=None):
+    def __init__(self, src, tstep=1e-3, duration=None, first_samp=0):
+        if duration is not None and duration < tstep:
+            raise ValueError('duration must be None or >= tstep.')
+        self.first_samp = first_samp
         self._src = src
         self._tstep = tstep
         self._labels = []
         self._waveforms = []
         self._events = np.empty((0, 3), dtype=int)
-        self._duration = duration
+        self._duration = duration  # if not None, sets # samples
         self._last_samples = []
         self._chk_duration = 1000
 
     @property
     def duration(self):
-        """Duration of the simulation."""
-        # If not, the precomputed maximum last sample is used
-        if self._duration is None:
-            return np.max(self._last_samples) * self._tstep
-        return self._duration
+        """Duration of the simulation in same units as tstep."""
+        if self._duration is not None:
+            return self._duration
+        return self.n_times * self._tstep
 
     @property
     def n_times(self):
         """Number of time samples in the simulation."""
-        return int(self.duration / self._tstep)
+        if self._duration is not None:
+            return int(self._duration / self._tstep)
+        ls = np.max(self._last_samples) if len(self._last_samples) > 0 \
+            else self.first_samp
```

Comment:
```suggestion
        ls = self.first_samp
        if len(self._last_samples) <= 0:
             ls = np.max(self._last_samples)
```
is more readable I think